src/fragmentarium/ui/SearchForm.test.tsx

```diff
@@ -5,9 +5,20 @@ import { factory } from 'factory-girl'
 import Promise from 'bluebird'
 import SessionContext from 'auth/SessionContext'
 import SearchForms from './SearchForm'
-import { createMemoryHistory } from 'history'
+import { createMemoryHistory, MemoryHistory } from 'history'
 import BibliographyEntry from 'bibliography/domain/BibliographyEntry'
 import userEvent from '@testing-library/user-event'
+import FragmentService from 'fragmentarium/application/FragmentService'
+import FragmentSearchService from 'fragmentarium/application/FragmentSearchService'
+import MemorySession from 'auth/Session'
+
+jest.mock('fragmentarium/application/FragmentService')
+
+const fragmentService = new (FragmentService as jest.Mock<FragmentService>)()
+const fragmentSearchService = new (FragmentSearchService as jest.Mock<
+  FragmentSearchService
+>)()
+const session = new (MemorySession as jest.Mock<MemorySession>)()
```

Comment:
Is it safe to have global test doubles? Changes made by one test could leak to the next.

Reply:
i can put everything withing a **describe block**.